libpod/container_log_linux.go

```diff
@@ -97,7 +97,16 @@ func (c *Container) readFromJournal(ctx context.Context, options *logs.LogOption
 	// exponential backoff.
 	var cursor string
 	var cursorError error
-	var containerCouldBeLogging bool
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
```

Comment:
Please unlock earlier.  The lock is only needed for the state check right below.  Just to be defensive for future changes which may perform more work after.

Reply:
fixed